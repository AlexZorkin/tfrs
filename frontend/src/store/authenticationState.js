import { put, takeLatest, all } from 'redux-saga/effects'
import { getLoggedInUser } from '../actions/userActions'
import { initKeycloak, loginKeycloakUserSuccess } from '../actions/keycloakActions'
import Keycloak from 'keycloak-js'
import ActionTypes from '../constants/actionTypes/Keycloak'
import configureAxios from './authorizationInterceptor'
import CONFIG from '../config'

export default function * authenticationStateSaga (store) {
  const { token, expiry } = store.getState().userAuth
  const now = Math.round(Date.now() / 1000)
  // Token already exists so setup things here
  if (token && now < expiry) {
    configureAxios()
    yield put(getLoggedInUser())
  }

  const keycloak = new Keycloak({
    url: CONFIG.KEYCLOAK.AUTH_URL,
    realm: CONFIG.KEYCLOAK.REALM,
    clientId: CONFIG.KEYCLOAK.CLIENT_ID
  })
  const authenticated = yield keycloak.init({
    pkceMethod: 'S256',
    redirectUri: CONFIG.KEYCLOAK.CALLBACK_URL,
    idpHint: 'idir'
  })

  yield put(initKeycloak(keycloak, authenticated))

  yield all([
    takeLatest(ActionTypes.LOGIN_KEYCLOAK_USER_SUCCESS, getBackendUser)
  ])

  if (authenticated) {
    yield put(loginKeycloakUserSuccess(keycloak.idToken, keycloak.idTokenParsed.exp))
  }
}

function * getBackendUser (action) {
  configureAxios(action.payload)
  yield put(getLoggedInUser())
}
