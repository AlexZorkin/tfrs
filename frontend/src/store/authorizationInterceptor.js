import axios from 'axios'
// import userManager from './oidc-usermanager';
import CONFIG from '../config'
import { keycloak } from '../actions/keycloakActions'

function configureAxios () {
  if (CONFIG.KEYCLOAK) {
    axios.interceptors.request.use((config) => {
      const loadconfig = async () => {
        // const user = await userManager.getUser();
        const kc = keycloak() // TODO see if this needs to be an async get
        // console.log("configureAxios user", kc)
        return {
          ...config,
          headers: {
            Authorization: `Bearer ${kc.idToken}`
          }
        }
      }

      return loadconfig()
    })
  }
}

export default configureAxios
