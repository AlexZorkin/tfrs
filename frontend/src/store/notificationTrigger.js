import { put, takeLatest, delay } from 'redux-saga/effects'
import { getNotifications, getNotificationsCount } from '../actions/notificationActions'
import UserActionTypes from '../constants/actionTypes/Users'
import NotificationActionTypes from '../constants/actionTypes/Notifications'

const TRIGGERING_ACTIONS = [
  'SERVER_INITIATED_NOTIFICATION_RELOAD',
  UserActionTypes.RECEIVE_LOGGED_IN_USER,
  NotificationActionTypes.SUCCESS_NOTIFICATIONS
]

function * fetchNotifications (store) {
  yield delay(1000) // debounce
  if (store.getState().rootReducer.userRequest.isAuthenticated &&
        store.getState().rootReducer.userRequest.loggedInUser?.id) {
    yield put(getNotificationsCount())
  }

  if (store.getState().rootReducer.notifications.onNotificationsPage) {
    yield put(getNotifications())
  }
}

export default function * notificationsSaga (store) {
  yield takeLatest(
    action => (TRIGGERING_ACTIONS.includes(action.type)),
    fetchNotifications,
    store
  )
}
