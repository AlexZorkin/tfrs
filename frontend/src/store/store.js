import createSagaMiddleware from 'redux-saga'
import { compose } from 'redux'
import thunk from 'redux-thunk'
import { configureStore } from '@reduxjs/toolkit'
import persistState from 'redux-localstorage'
import { createLogger } from 'redux-logger'
import createSocketIoMiddleware from 'redux-socket.io'
import { reducer as toastrReducer } from 'react-redux-toastr'
import io from 'socket.io-client'

import rootReducer from '../reducers/reducer'
import { SOCKETIO_URL } from '../constants/routes'

import { persistTargetPathReducer } from '../reducers/persistTargetPathReducer'

import CONFIG from '../config'
// import sessionTimeoutSaga from './sessionTimeout'
import authenticationStateSaga from './authenticationState'
import notificationsSaga from './notificationTrigger'
import socketAuthenticationSaga from './socketAuthentication'
import autocompleteInvalidatorSaga from './autocompleteInvalidator'
import { carbonIntensities } from '../actions/carbonIntensities'
import { defaultCarbonIntensities } from '../actions/defaultCarbonIntensities'
import { energyDensities } from '../actions/energyDensities'
import { energyEffectivenessRatios } from '../actions/energyEffectivenessRatios'
import { expectedUses } from '../actions/expectedUses'
import { fuelClasses } from '../actions/fuelClasses'
import { notionalTransferTypes } from '../actions/notionalTransferTypes'
import { petroleumCarbonIntensities } from '../actions/petroleumCarbonIntensities'
import { transactionTypes } from '../actions/transactionTypes'
import { roles } from '../actions/roleActions'
import autosaveSaga from './autosaveStore'
import { complianceReporting } from '../actions/complianceReporting'
import { exclusionReports } from '../actions/exclusionReports'

const sagaMiddleware = createSagaMiddleware()

const socket = io(SOCKETIO_URL)
const socketIoMiddleware = createSocketIoMiddleware(socket, 'socketio/')

const enhancer = compose(persistState(['targetPath'], { key: 'tfrs-state' }))

const combinedReducers = (state = {}, action) => {
  const currentRoute = state.routing || {}
  return {
    toastr: toastrReducer(state.toastr, action),
    targetPath: persistTargetPathReducer(state.targetPath, { ...action, currentRoute }),
    rootReducer: rootReducer(state.rootReducer, action)
  }
}

const allMiddleware = [
  socketIoMiddleware,
  sagaMiddleware,
  thunk
]

if (CONFIG.DEBUG.ENABLED) {
  allMiddleware.push(createLogger())
}

const store = configureStore({
  reducer: combinedReducers,
  middleware: [...allMiddleware],
  enhancers: [enhancer]
})

// sagaMiddleware.run(sessionTimeoutSaga); // TODO fix sessiontimeout with new keycloak login
sagaMiddleware.run(notificationsSaga, store)
sagaMiddleware.run(authenticationStateSaga, store)
sagaMiddleware.run(socketAuthenticationSaga, store)
sagaMiddleware.run(autocompleteInvalidatorSaga, store)
sagaMiddleware.run(autosaveSaga)

sagaMiddleware.run(roles.saga)
sagaMiddleware.run(complianceReporting.saga)
sagaMiddleware.run(exclusionReports.saga)
sagaMiddleware.run(carbonIntensities.saga)
sagaMiddleware.run(defaultCarbonIntensities.saga)
sagaMiddleware.run(energyDensities.saga)
sagaMiddleware.run(energyEffectivenessRatios.saga)
sagaMiddleware.run(expectedUses.saga)
sagaMiddleware.run(fuelClasses.saga)
sagaMiddleware.run(notionalTransferTypes.saga)
sagaMiddleware.run(petroleumCarbonIntensities.saga)
sagaMiddleware.run(transactionTypes.saga)

export default store
