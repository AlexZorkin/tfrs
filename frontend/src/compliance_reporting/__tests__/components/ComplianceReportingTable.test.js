import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer'
import ComplianceReportingTable from "../../components/ComplianceReportingTable";
import store from "../../../store/store";

describe('ComplianceReportingTable', ()=> {
    test('should render the component', ()=> {
        const props = {
            loggedInUser : {
                isGovernmentUser: true
            },
            items : [],
            isFetching:true,
            navigate : jest.fn(),
            itemsCount:0,
            isEmpty: true,
            isFetching: false,
            getComplianceReports: jest.fn()
        }
        const component = renderer.create(
            <BrowserRouter>
                <Provider store={store}>
                    <ComplianceReportingTable {...props} />
                </Provider>
            </BrowserRouter>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})