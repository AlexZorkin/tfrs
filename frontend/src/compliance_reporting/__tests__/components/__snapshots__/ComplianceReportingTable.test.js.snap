// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplianceReportingTable should render the component 1`] = `
<div
  className="ReactTable searchable complianceReportListTable"
  style={{}}
>
  <div
    className="rt-table"
    role="grid"
  >
    <div
      className="rt-thead -header"
      style={
        {
          "minWidth": "520px",
        }
      }
    >
      <div
        className="rt-tr"
        role="row"
      >
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "50 0 auto",
              "maxWidth": null,
              "width": "50px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Compliance Period
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Organization
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Display Name
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Original Status
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Supplemental Status
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Current Status
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "95 0 auto",
              "maxWidth": null,
              "width": "95px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Last Updated On
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="rt-thead -filters"
      style={
        {
          "minWidth": "520px",
        }
      }
    >
      <div
        className="rt-tr"
        role="row"
      >
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "50 0 auto",
              "maxWidth": null,
              "width": "50px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "75 0 auto",
              "maxWidth": null,
              "width": "75px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            {
              "flex": "95 0 auto",
              "maxWidth": null,
              "width": "95px",
            }
          }
          tabIndex="-1"
        >
          <input
            onChange={[Function]}
            style={
              {
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="rt-tbody"
      style={
        {
          "minWidth": "520px",
        }
      }
    >
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -odd"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -even"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -odd"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -even"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -odd"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -even"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -odd"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -even"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -odd"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -padRow -even"
          role="row"
          style={{}}
        >
          <div
            className="rt-td col-compliance-year"
            role="gridcell"
            style={
              {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-organization"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-displayname"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-supplemental-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-status"
            role="gridcell"
            style={
              {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
          >
            <span>
               
            </span>
          </div>
          <div
            className="rt-td col-date"
            role="gridcell"
            style={
              {
                "flex": "95 0 auto",
                "maxWidth": null,
                "width": "95px",
              }
            }
          >
            <span>
               
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pagination-bottom"
  >
    <div
      className="-pagination"
    >
      <div
        className="-previous"
      >
        <button
          className="-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Previous
        </button>
      </div>
      <div
        className="-center"
      >
        <span
          className="-pageInfo"
        >
          Page
           
          <div
            className="-pageJump"
          >
            <input
              aria-label="jump to page"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              type="number"
              value={1}
            />
          </div>
           
          of
           
          <span
            className="-totalPages"
          >
            1
          </span>
        </span>
        <span
          className="select-wrap -pageSizeOptions"
        >
          <select
            aria-label="rows per page"
            onChange={[Function]}
            value={10}
          >
            <option
              value={5}
            >
              5 rows
            </option>
            <option
              value={10}
            >
              10 rows
            </option>
            <option
              value={15}
            >
              15 rows
            </option>
            <option
              value={20}
            >
              20 rows
            </option>
            <option
              value={25}
            >
              25 rows
            </option>
            <option
              value={50}
            >
              50 rows
            </option>
            <option
              value={100}
            >
              100 rows
            </option>
          </select>
        </span>
      </div>
      <div
        className="-next"
      >
        <button
          className="-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
  <div
    className="rt-noData"
  >
    No rows found
  </div>
  <div
    className="-loading"
  >
    <div
      className="-loading-inner"
    >
      Loading...
    </div>
  </div>
</div>
`;
